src/conv/invokers/impl_gemm.cpp

```diff
@@ -0,0 +1,147 @@
+                                        "xdlops_nchw_kcyx_nkhw" ||
+                    kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_"
+                                        "xdlops_gnchw_gkcyx_gnkhw" ||
+                    kernel.GetName() ==
+                        "gridwise_convolution_backward_data_implicit_gemm_v1r1_nchw_kcyx_nkhw"))
+                {
+                    float zero = 0.f;
+                    TensorDescriptor workspaceDesc(
+                        miopenFloat, tensors.outDesc.GetLengths(), tensors.outDesc.GetStrides());
+                    SetTensor(handle, workspaceDesc, workSpace, &zero);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+
+                    kernel(tensors.in, tensors.w, workSpace);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+
+                    CastTensor(handle,
+                               &lowp_quant,
+                               workspaceDesc,
+                               workSpace,
+                               tensors.outDesc,
+                               tensors.out,
+                               0,
+                               0);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+                }
+                else if((kernel.GetName() == "gridwise_convolution_implicit_gemm_v4_nchw_kc1x1_"
+                                             "nkhw_lds_double_buffer") ||
+                        (kernel.GetName() ==
+                         "gridwise_convolution_implicit_gemm_v4r4_xdlops_nchw_kc1x1_"
+                         "nkhw_lds_double_buffer"))
+                {
+                    bool hasStride =
+                        (tensors.inDesc.GetLengths()[2] != tensors.outDesc.GetLengths()[2]) ||
+                        (tensors.inDesc.GetLengths()[3] != tensors.outDesc.GetLengths()[3]);
+                    /// \todo set zero within implicitGEMM kernel
+                    if(hasStride)
+                    {
+                        MIOPEN_LOG_I2("hasStride, call SetTensor with zero");
+                        float zero = 0.f;
+                        SetTensor(handle, tensors.outDesc, tensors.out, &zero);
+
+                        if(handle.IsProfilingEnabled())
+                            elapsed += handle.GetKernelTime();
+                    }
+
+                    kernel(tensors.in, tensors.w, tensors.out);
+
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+                }
+                else if(kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_"
+                                            "nchw_kcyx_nkhw" ||
+                        kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_"
```

Comment:
name split

Reply:
@DrizztDoUrden this split is a leftover of clang-format that impedes the programmers from finding kernel names in the code. Please restore strings to undivided state.

@asroy We use ridiculously long names for implicit gemm and I have no idea why. Can we rework filenames and kernel names at once?

Reply:
@asroy I mean this `gridwise_convolution_backward_data_implicit_gemm` is always the same (except "backward_data" of course), so I would replace it with smth like `gw_conv_fwd/bwd/wrw_igemm` or even `conv_igemm_gw_fwd/bwd/wrw`...

Reply:
> Please restore strings to undivided state.

Oops, already done, missed to reload page.